Makefile

```diff
@@ -69,10 +75,22 @@ libp2p_helper:
 # Alias
 dht: kademlia libp2p_helper
 
-build: git_hooks reformat-diff
+_build/default/src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe:
+	$(info Building runtime_genesis_ledger)
+	ulimit -s 65532 && (ulimit -n 10240 || true) && $(WRAPAPP) env CODA_COMMIT_SHA1=$(GITLONGHASH) dune build --profile=$(DUNE_PROFILE) src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe src/app/runtime_genesis_ledger/genesis_filename.txt
+.PHONY: _build/default/src/app/runtime_genesis_ledger/runtime_genesis_ledger.exe
+
+#Genesis state autogen path
+genesis_path:
+GENESIS_STATE_PATH = $(CODA_TMP_DIR)/coda_cache_dir/$(shell cat _build/default/src/app/runtime_genesis_ledger/genesis_filename.txt).tar.gz
```

Comment:
This will work only if this txt file exists before you run make since this `$(shell ...)` expression is evaluated and expanded during preprocessing.

Reply:
it should exists because I added it as dependency in the target `genesis_ledger`? (I assumed dependencies are checked in sequence) If not, then i'll have to add the exe as a dependency to the genesis_filename target?

Reply:
It won't exist yet unless a previous build already created it. The preprocessor is run completely before any targets are built, so even if it's in the dependency graph of the target you want make to build, this would resolve before that dependency is built.